internal/services/network/custom_ip_prefix_resource.go

```diff
@@ -0,0 +1,419 @@
+		ContinuousTargetOccurence: 3,
+	}
+
+	if d.IsNewResource() {
+		stateConf.Timeout = d.Timeout(pluginsdk.TimeoutCreate)
+	} else {
+		stateConf.Timeout = d.Timeout(pluginsdk.TimeoutUpdate)
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("waiting for %s to become ready: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceCustomIpPrefixRead(d, meta)
+}
+
+func customIpPrefixCreateRefreshFunc(ctx context.Context, client *network.CustomIPPrefixesClient, id parse.CustomIpPrefixId) pluginsdk.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.Get(ctx, id.ResourceGroup, id.CustomIpPrefixeName, "")
+		if err != nil {
+			if utils.ResponseWasNotFound(res.Response) {
+				return nil, "NotFound", nil
+			}
+			return nil, "", fmt.Errorf("polling for the commissionedstate of the Custom Ip Prefix %q (Resource Group: %q): %+v", id.CustomIpPrefixeName, id.ResourceGroup, err)
+		}
+
+		if res.CustomIPPrefixPropertiesFormat == nil {
+			return nil, "", fmt.Errorf("unexpected nil properties format of Custom Ip Prefix %q (Resource Group %q)", id.CustomIpPrefixeName, id.ResourceGroup)
+		}
+		return res, string(res.CustomIPPrefixPropertiesFormat.CommissionedState), nil
+	}
+}
+
+func resourceCustomIpPrefixRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.CustomIPPrefixesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CustomIpPrefixID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.CustomIpPrefixeName, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] %s does not exist - removing from state", id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("location", azure.NormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", id.ResourceGroup)
+
+	if props := resp.CustomIPPrefixPropertiesFormat; props != nil {
+		d.Set("cidr", props.Cidr)
+		d.Set("authorization_message", props.AuthorizationMessage)
+		d.Set("signed_message", props.SignedMessage)
+	}
+
+	if v := resp.Zones; v != nil {
+		d.Set("zones", zones.Flatten(v))
+	}
+
+	// The "action" will be changed by API automatically, hence setting it from config.
+	d.Set("action", d.Get("action").(string))
+
+	if err := d.Set("tags", tags.Flatten(resp.Tags)); err != nil {
+		return fmt.Errorf("setting `tags`: %+v", err)
+	}
+
+	return nil
+}
+
+func resourceCustomIpPrefixDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.CustomIPPrefixesClient
+	id, err := parse.CustomIpPrefixID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := prepareDelete(d, meta, *id); err != nil {
+		return err
+	}
+
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
```

Comment:
this timeout is being set within the `prepareDelete` method too - so by this point in time the timeout won't be correct, we'll need to pass `ctx` into `prepareDelete` (or remove that method)

Reply:
Fixed.